logstash-core/src/main/java/org/logstash/ext/JrubyTimestampExtLibrary.java

```diff
@@ -301,6 +310,21 @@ public IRubyObject minus(final ThreadContext context, final IRubyObject val) {
             );
         }
 
+        /**
+         * Returns the number of nanoseconds since the unix epoch.
+         *
+         * This value is only sensible for the years 1678-2262 CE.
+         *
+         * @return
```

Comment:
I have refactored to avoid this edge-case. There were two methods that relied on it:

Ruby `LogStash::Timestamp#to_f`: The resulting value loses precision as we get sufficiently far from the epoch, but the math has been re-arranged to avoid a possible overflow
Ruby `LogStash::Timestamp#time`: uses `RubyTime.newTime(Runtime, org.joda.time.DateTime, long)` to provide a millis-granularity date-time and excess nanos, thereby routing around the potential overflow.